src/catalog/src/durable/impls/persist/metrics.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Prometheus monitoring metrics.
+
+use mz_ore::metric;
+use mz_ore::metrics::{IntCounter, MetricsRegistry};
+use mz_ore::stats::histogram_seconds_buckets;
+use prometheus::{Histogram, IntCounterVec};
+
+#[derive(Debug, Clone)]
+pub struct Metrics {
+    pub transactions: IntCounter,
+    pub transaction_commit_errors: IntCounterVec,
+    pub transaction_commit_latency_duration_seconds: Histogram,
+    pub snapshot_latency_duration_seconds: Histogram,
+    pub sync_latency_duration_seconds: Histogram,
+}
+
+impl Metrics {
+    /// Returns a new [Metrics] instance connected to the given registry.
+    pub fn new(registry: &MetricsRegistry) -> Self {
+        Self {
+            transactions: registry.register(metric!(name: "catalog_transactions", help: "Total number of started transactions.")),
```

Comment:
nit: I don't think rustfmt can understand macros, so it won't do this for you, but I'd put one thing per line here (like you have below) to make it a bit easier to scan

Reply:
Done